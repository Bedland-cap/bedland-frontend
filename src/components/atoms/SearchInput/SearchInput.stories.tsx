import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import SearchInput from './SearchInput';

const meta = {
  title: 'atoms/SearchInput',
  component: SearchInput,
  tags: ['autodocs'],
  parameters: {
    backgrounds: { default: 'light' },
  },
  argTypes: {
    placeholder: { control: 'text', description: 'Placeholder text in the search input' },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof SearchInput>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    placeholder: 'Search buildings...',
  },
};
