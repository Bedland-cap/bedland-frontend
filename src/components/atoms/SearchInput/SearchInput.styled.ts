import styled from 'styled-components';
import { HexColor, PaletteType } from 'theme/theme.types';

export const InputContainer = styled.div<{ outlineColor: HexColor }>`
  align-items: center;
  background: #fff;
  border-radius: 8px;
  display: flex;
  height: 2.5rem;
  padding: 0.5rem;
  width: 17.5rem;

  :focus-within {
    outline: ${({ outlineColor }) => `1px solid ${outlineColor}`};
  }
`;

export const Input = styled.input<{ placeholder: string; palette: PaletteType }>`
  background-color: transparent;
  color: ${({ palette }) => palette.inputGrey};
  height: 1.75rem;
  width: 15rem;

  ::placeholder {
    color: ${({ palette }) => palette.greyDark};
  }
`;

export const IconWrapper = styled.div`
  align-items: center;
  display: flex;
  margin-left: 0.583rem;
  margin-right: 0.771rem;
`;
