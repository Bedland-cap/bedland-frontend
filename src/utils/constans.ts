export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
