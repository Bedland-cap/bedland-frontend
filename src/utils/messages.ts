const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
