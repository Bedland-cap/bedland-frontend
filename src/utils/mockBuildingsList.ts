import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
