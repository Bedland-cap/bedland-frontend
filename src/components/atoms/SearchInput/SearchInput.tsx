import { useContext, useState } from 'react';
import { ThemeContext } from 'theme/ThemeContext';
import Icon from '../Icon/Icon';
import * as Styled from './SearchInput.styled';

const SearchInput = ({ placeholder }: { placeholder: string }) => {
  const [hasFocus, setFocus] = useState(false);
  const { palette } = useContext(ThemeContext);

  return (
    <Styled.InputContainer outlineColor={palette.primaryDark}>
      <Styled.IconWrapper>
        <Icon
          name='search'
          size={16}
          color={hasFocus ? 'primaryDark' : 'greyDark'}
          isActive={false}
        />
      </Styled.IconWrapper>
      <Styled.Input
        placeholder={placeholder}
        palette={palette}
        onFocus={() => setFocus(true)}
        onBlur={() => setFocus(false)}
      />
    </Styled.InputContainer>
  );
};

export default SearchInput;
