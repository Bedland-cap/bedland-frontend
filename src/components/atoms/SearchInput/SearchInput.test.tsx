import { renderWithProviders } from 'utils/test.utils';
import SearchInput from './SearchInput';

describe('SearchInput', () => {
  it('renders correctly', () => {
    const input = renderWithProviders(<SearchInput placeholder='Search buildings...' />, {});
    expect(input).toMatchSnapshot();
  });
});
