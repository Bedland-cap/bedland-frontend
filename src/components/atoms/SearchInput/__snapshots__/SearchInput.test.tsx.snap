// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe eCLQnL"
      >
        <div
          class="sc-dkrFOg fGrSpP"
        >
          <svg
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-search"
            />
            <path
              d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
              stroke="#767676"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.15972"
            />
          </svg>
        </div>
        <input
          class="sc-gswNZR ffqmbe"
          placeholder="Search buildings..."
        />
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe eCLQnL"
    >
      <div
        class="sc-dkrFOg fGrSpP"
      >
        <svg
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-search"
          />
          <path
            d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
            stroke="#767676"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.15972"
          />
        </svg>
      </div>
      <input
        class="sc-gswNZR ffqmbe"
        placeholder="Search buildings..."
      />
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
